node/src/main/scala/org/bitcoins/node/NeutrinoNode.scala

```diff
@@ -36,11 +36,11 @@ case class NeutrinoNode(
 
   implicit override def nodeAppConfig: NodeAppConfig = nodeConfig
 
-  override def chainAppConfig: ChainAppConfig = chainConfig
+  implicit override def chainAppConfig: ChainAppConfig = chainConfig
 
-  val controlMessageHandler: ControlMessageHandler = ControlMessageHandler(this)
+  implicit def node: NeutrinoNode = this
```

Comment:
Lets not get crazy with `implicit`. Its easy to overuse `implicit` and then your code becomes very unreadable quickly.

Our general policy is its ok to use `implicit` for things like `ActorSystem`, `AppConfig` (generally any configuration related code), but not for normal data structures like `NeutrinoNode` or `Wallet` etc.